src/components/WatchModal.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Modal } from 'react-bootstrap';
+import watchService, { WatchMapping } from 'services/watchService';
+import { MdDelete } from 'react-icons/md';
+import { HiArrowNarrowRight } from 'react-icons/hi';
+
+function WatchModal({ watchModalView, setWatchModalView }) {
+    const [mappings, setMappings] = useState<WatchMapping[]>([]);
+    const [shouldUpdateMappings, setShouldUpdateMappings] = useState(true);
+    const [inputFolderPath, setInputFolderPath] = useState('');
+    const [inputCollectionName, setInputCollectionName] = useState('');
+
+    useEffect(() => {
+        if (shouldUpdateMappings) {
+            setMappings(watchService.getWatchMappings());
+            setShouldUpdateMappings(false);
+        }
+    }, [shouldUpdateMappings]);
+
+    const handleFolderSelection = async () => {
+        const folderPath = await watchService.selectFolder();
+        setInputFolderPath(folderPath);
+    };
+
+    const handleCollectionNameChange = (
+        e: React.ChangeEvent<HTMLInputElement>
+    ) => {
+        setInputCollectionName(e.target.value);
+    };
+
+    const handleAddWatchMapping = async () => {
+        if (inputFolderPath.length > 0 && inputCollectionName.length > 0) {
+            await watchService.addWatchMapping(
+                inputCollectionName,
+                inputFolderPath
+            );
+            setInputCollectionName('');
+            setInputFolderPath('');
+            setShouldUpdateMappings(true);
+        }
+    };
+
+    const handleRemoveWatchMapping = async (mapping: WatchMapping) => {
+        await watchService.removeWatchMapping(mapping.collectionName);
+        setShouldUpdateMappings(true);
+    };
+
+    return (
+        <Modal
```

Comment:
Will refactor this into components after finalizing on design, currently it looks very basic.